pkg/controller/controlplane/valuesprovider.go

```diff
@@ -499,37 +547,37 @@ func getCalicoBgpChartValues(
 	var serviceLbIPs, serviceExtIPs, serviceClusterIPs []string
 	var peers []map[string]any
 	var filters []map[string]any
-	if cpConfig.LoadBalancerConfig.CalicoBgpConfig != nil &&
+	if cpConfig.LoadBalancerConfig.CalicoConfig == nil || cpConfig.LoadBalancerConfig.CalicoConfig.CalicoBgpConfig != nil &&
 		*cluster.Shoot.Spec.Networking.Type == metal.ShootCalicoNetworkType {
-		if cpConfig.LoadBalancerConfig.CalicoBgpConfig.ServiceLoadBalancerIPs != nil {
-			for _, cidr := range cpConfig.LoadBalancerConfig.CalicoBgpConfig.ServiceLoadBalancerIPs {
+		if cpConfig.LoadBalancerConfig.CalicoConfig.CalicoBgpConfig.ServiceLoadBalancerIPs != nil {
```

Comment:
Could `cpConfig.LoadBalancerConfig.CalicoConfig.CalicoBgpConfig` be bound to a variable after checking it is not nil to shorten the uses in the rest of the function?

Reply:
good idea to reduce clutter. done